src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs

```diff
@@ -485,6 +486,33 @@ private void SetOptionValue<T>(string name, T value)
             {
                 Options.Add(name, value);
             }
-        }
+        }
+
+        public bool IsComparableTo(ConnectionDetails other)

```

Comment:
Yeah, I figured it was something like this. It's not a direct 1:1 comparison of the two objects, so it's not quite accurate to call it IEquatable. And since there's no real ordering that can be done with these, it doesn't make sense to use IComparable. Good to know we're on the same page with this.